jcommune-view/jcommune-web-view/src/main/webapp/resources/javascript/app/userSearch.js

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2011  JTalks.org Team
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+var userSearch = {};
+
+userSearch.popup = {};
+
+userSearch.popup.show = function(message) {
+    var alertError = $(".alert-error");
+    alertError.show();
+    alertError.find('#alertMessagePopup').text(message);
+
+    alertError.find('.close').on('click', function() {
+        alertError.hide();
+    });
+};
+
+userSearch.connectionErrorCallback = function(xhr, status) {
+    if (status == 'timeout' || xhr.status == 0) {
+        if (xhr.status == 0) {
+            // Need it because firefox makes no difference between refused connection and
+            // aborted request
+            setTimeout(function () {
+                userSearch.popup.show($labelUserSearchConnectionLost);
+            }, 3000);
+        } else {
+            userSearch.popup.show($labelUserSearchConnectionLost);
+        }
+    } else if (xhr.status == 403) {
+        userSearch.popup.show($labelNotLoggedInError);
+    }
+};
+
+userSearch.groupSelected = function(userID, groupID) {
+    $.ajax({
+        url: $root + "/user/" + userID + "/group/" + groupID,
+        type: 'POST',
+        success: function(result) {
+            if (result.status != 'SUCCESS') {
+                userSearch.popup.show($labelUserGroupAddError);
+            }
+        }
+    }).fail(userSearch.connectionErrorCallback);
+};
+
+userSearch.groupDeleted = function(userID, groupID) {
+    $.ajax({
+        url: $root + "/user/" + userID + "/group/" + groupID,
+        type: 'DELETE',
+        success: function(result) {
+            if (result.status != 'SUCCESS') {
+                userSearch.popup.show($labelUserGroupDeleteError);
+            }
+        }
+    }).fail(userSearch.connectionErrorCallback);
+};
+
+userSearch.toggleUserGroups = function(event, userID) {
+    $('#user-groups-table-' + userID).toggle(0, function() {
+        if($(this).is(":visible")) {
+            userSearch.showUserGroups(userID);
+        }
+    }).fail(userSearch.connectionErrorCallback);
```

Comment:
This line fails each time I press Groups:
`Uncaught TypeError: $(...).toggle(...).fail is not a function`